sync/src/synchronizer/mod.rs

```diff
@@ -51,27 +52,60 @@ struct BlockFetchCMD {
     sync: Synchronizer,
     p2p_control: ServiceControl,
     recv: channel::Receiver<FetchCMD>,
+    can_start: bool,
+    number: BlockNumber,
 }
 
 impl BlockFetchCMD {
-    fn run(&self) {
+    fn run(&mut self) {
         while let Ok(cmd) = self.recv.recv() {
             match cmd {
                 FetchCMD::Fetch(peers) => {
-                    for peer in peers {
-                        if let Some(fetch) =
-                            BlockFetcher::new(&self.sync, peer, IBDState::In).fetch()
-                        {
-                            for item in fetch {
-                                BlockFetchCMD::send_getblocks(item, &self.p2p_control, peer);
+                    if self.can_start() {
+                        for peer in peers {
+                            if let Some(fetch) =
+                                BlockFetcher::new(&self.sync, peer, IBDState::In).fetch()
+                            {
+                                for item in fetch {
+                                    BlockFetchCMD::send_getblocks(item, &self.p2p_control, peer);
+                                }
                             }
                         }
+                    } else {
+                        let best_known = self.sync.shared.state().shared_best_header_ref();
+                        let number = best_known.number();
+                        if number != self.number && number % 10000 == 0 {
+                            self.number = number;
+                            info!(
```

Comment:
Does it worth a metric record?

Reply:
This only appears when synchronizing data from the beginning, and will not happen again, I think it should not be necessary to use metric @yangby-cryptape @keroro520

Reply:
I agree with @driftluo.

Reply:
The floor of memory limit is increased when synchronizing data from the beginning.